CXX-3008 Add post-release instructions for Silk, Snyk, and patch release tags (#1148)
mongodb/mongo-cxx-driver: etc/releasing.md

  GRS_CONFIG_USER1_USERNAME=<username>
  GRS_CONFIG_USER1_PASSWORD=<password>
  ```
- Snyk credentials. Save these to `~/.secrets/snyk-creds.txt`:
  ```bash
  SNYK_API_TOKEN=<token>

>   ```

Reviewer: Suggest adding the `SILK_CLIENT_ID` and `SILK_CLIENT_SECRET` needed for "Update Silk" steps below.

Author: Done.

The new branch should be continuously tested on Evergreen. Update the "Display Name" and "Branch Name" of the [mongo-cxx-driver-latest-release Evergreen project](https://spruce.mongodb.com/project/mongo-cxx-driver-latest-release/settings/general) to refer to the new release branch.

## Update Silk and Snyk with new branch if necessary

After creating the new minor release branch in the prior step, update Silk and Snyk to trach the new release branch.

> For Silk, see the [silk-create-asset-group.sh script](https://github.com/mongodb/libmongocrypt/blob/master/etc/silk-create-asset-group.sh) in the C Driver as reference for the commands to run against the Silk API. Use `mongo-cxx-driver` as the name and prefix in place of `mongo-c-driver` accordingly.

Reviewer: Step and PR description refers to "C driver" but links to libmongocrypt. If C driver is intended, suggest updating:

```suggestion
For Silk, use the [create-silk-asset-group.py script](https://github.com/mongodb/mongo-c-driver/blob/master/tools/create-silk-asset-group.py) in the C Driver to create a new Silk asset group.
```

Author: Good catch.
